gprofiler/main.py

```diff
@@ -319,30 +324,39 @@ def _snapshot(self) -> None:
             )
             raise
         metadata = (
-            get_current_metadata(cast(Metadata, self._static_metadata))
+            get_current_metadata(cast(ProfileMetadata, self._static_metadata))
             if self._collect_metadata
             else {"hostname": get_hostname()}
         )
         metadata.update({"profiling_mode": self._profiler_state.profiling_mode})
         metrics = self._system_metrics_monitor.get_metrics()
+
+        try:
+            external_app_metadata = read_external_metadata(self._external_metadata_path).application
+        except ExternalMetadataStaleError:
+            logger.warning("External metadata is stale, ignoring it")
```

Comment:
you find it useful? won't it just spam?

Reply:
If occasionally the file is not updated - it will help us find bugs. If the file is never updated - yeah, it'll spam. But I'm more afraid of the first case, so keeping it for now